### PR TITLE
Migrated plugins from request and http to axios client

### PR DESCRIPTION
**Pull request by SloCompTech**
# Description

Migrated all plugins that use either request or pure http/https to make request to other webservices to [Axios library](https://github.com/axios/axios), which handles http/https for you (no need for https and http to call apis differently) and "detects" content type and handles content length header.

Updated plugins:

- SimpleWebook
- MessageRepeat
- AdvancedWebook
- MicrosoftTeams
- Gotify

Fixes:
- Message Repeat: `Content-Type` written `content-type`, but it should be `Content-Type`. This causes `Content-Type` header to default to `application/x-www-form-urlencoded` instead of `application/json` #399 
- Advanced WebHook: No header `Content-Length` which is required for `Content-Type: application/json` to work properly on other eg. PHP web services. #400 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

SimpleWebook plugin:

- external API: OK

```
[pmon]  2020-05-05 16:19:11  debug : plugin: SimpleWebhook
[pmon]  2020-05-05 16:19:11  debug : RUNNING PLUGIN!
[pmon]  2020-05-05 16:19:11  debug : SimpleWebhook: Sending to https://example.org/api/mesages: {"address":"0000000","message":"TEST 2HOP FORWARD 4 time","source":"test"}
[pmon]  2020-05-05 16:19:11  debug : { id: 14,
  address: '0000000',
  message: 'TEST 2HOP FORWARD 4 time',
  source: 'test',
  timestamp: 1588688350,
  alias_id: null,
  alias: null,
  agency: null,
  icon: null,
  color: null,
  ignore: null,
  pluginconf: {},
  pluginData: {} }
[pmon]  2020-05-05 16:19:11  debug : afterMessage done
[pmon]  2020-05-05 16:19:11  info : SimpleWebhook: Message Sent
```

MessageRepeat plugin:

- To other Pagermon Instance: OK
- To external API: OK

```
[pmon]  2020-05-05 15:43:08  debug : MessageRepeat: Sending to https://ptsv2.com/t/XXX/post: {"address":"0000000","message":"TEST PAGERMON NEW 19","source":"test","datetime":"1588686177","UUID":"notimportant"}
[pmon]  2020-05-05 15:43:08  debug : ======================
[pmon]  2020-05-05 15:43:08  debug : plugin: AdvancedWebhook
[pmon]  2020-05-05 15:43:08  debug : { address: '0000000',
  datetime: '1588686177',
  message: 'TEST PAGERMON NEW 19',
  source: 'test',
  pluginData: {} }
[pmon]  2020-05-05 15:43:08  debug : beforeMessage done
[pmon]  2020-05-05 15:43:09  debug : afterMessage start
[pmon]  2020-05-05 15:43:09  debug : ======================
[pmon]  2020-05-05 15:43:09  debug : trigger: message scope: after
[pmon]  2020-05-05 15:43:09  debug : ======================
[pmon]  2020-05-05 15:43:09  debug : data object
[pmon]  2020-05-05 15:43:09  debug : { id: 9,
  address: '0000000',
  message: 'TEST PAGERMON NEW 19',
  source: 'test',
  timestamp: 1588686177,
  alias_id: null,
  alias: null,
  agency: null,
  icon: null,
  color: null,
  ignore: null,
  pluginconf: {},
  pluginData: {} }
[pmon]  2020-05-05 15:43:09  debug : plugins object
[pmon]  2020-05-05 15:43:09  debug : { MessageRepeat:
   { enable: true,
     repeatURI: 'https://ptsv2.com/t/XXX/post',
     repeatAPIKEY: 'notimportant',
     repeatUUID: 'notimportant' },
  AdvancedWebhook:
   { enable: false,
     URL: 'https://ptsv2.com/t/XXX/post',
     contentType: 'application/json',
     filterMode: { value: '2', text: 'All messages' },
     contentTemplate:
      '{"address": "/address/", "datetime": "/timestamp/", "message": "/message/", "source": "/source/"}' } }
[pmon]  2020-05-05 15:43:09  debug : ======================
[pmon]  2020-05-05 15:43:09  debug : ======================
[pmon]  2020-05-05 15:43:09  debug : plugin: MessageRepeat
[pmon]  2020-05-05 15:43:09  debug : Plugin does not run in this scope
[pmon]  2020-05-05 15:43:09  debug : ======================
[pmon]  2020-05-05 15:43:09  debug : plugin: AdvancedWebhook
[pmon]  2020-05-05 15:43:09  debug : { id: 9,
  address: '0000000',
  message: 'TEST PAGERMON NEW 19',
  source: 'test',
  timestamp: 1588686177,
  alias_id: null,
  alias: null,
  agency: null,
  icon: null,
  color: null,
  ignore: null,
  pluginconf: {},
  pluginData: {} }
[pmon]  2020-05-05 15:43:09  debug : afterMessage done
::1 - - [05/May/2020:13:43:09 +0000] "POST /api/messages HTTP/1.1" 200 1 "-" "PostmanRuntime/7.24.1"
[pmon]  2020-05-05 15:43:09  info : MessageRepeat: Message Sent
```
AdvancedWebhook plugin:

- External API: OK

```
[pmon]  2020-05-05 14:03:31  debug : AdvancedWebhook: Sending to https://ptsv2.com/t/XXX/post: {"address": "0000000", "datetime": "1588680234", "message": "TEST PAGERMON NEW 16", "source": "test"}
[pmon]  2020-05-05 14:03:31  debug : { id: 6,
  address: '0000000',
  message: 'TEST PAGERMON NEW 16',
  source: 'test',
  timestamp: 1588680234,
  alias_id: null,
  alias: null,
  agency: null,
  icon: null,
  color: null,
  ignore: null,
  pluginconf: {},
  pluginData: {} }
[pmon]  2020-05-05 14:03:31  debug : afterMessage done
::1 - - [05/May/2020:12:03:31 +0000] "POST /api/messages HTTP/1.1" 200 1 "-" "PostmanRuntime/7.24.1"
[pmon]  2020-05-05 14:03:32  info : AdvancedWebhook: Message Sent
```

Example error stacktrace:

```
[pmon]  2020-05-05 14:07:22  debug : AdvancedWebhook: Sending to https://example.com/t/XXX/post: {"address": "0000000", "datetime": "1588680301", "message": "TEST PAGERMON NEW 18", "source": "test"}
[pmon]  2020-05-05 14:07:22  debug : { id: 8,
  address: '0000000',
  message: 'TEST PAGERMON NEW 18',
  source: 'test',
  timestamp: 1588680301,
  alias_id: null,
  alias: null,
  agency: null,
  icon: null,
  color: null,
  ignore: null,
  pluginconf: {},
  pluginData: {} }
[pmon]  2020-05-05 14:07:22  debug : afterMessage done
::1 - - [05/May/2020:12:07:22 +0000] "POST /api/messages HTTP/1.1" 200 1 "-" "PostmanRuntime/7.24.1"
[pmon]  2020-05-05 14:07:23  error : AdvancedWebhook: Headers: {"cache-control":"max-age=604800","content-type":"text/html; charset=UTF-8","date":"Tue, 05 May 2020 12:07:22 GMT","expires":"Tue, 12 May 2020 12:07:22 GMT","server":"EOS (vny/0452)","content-length":"445","connection":"close"}
[pmon]  2020-05-05 14:07:23  error : AdvancedWebhook: Data: <?xml version="1.0" encoding="iso-8859-1"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
         "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
        <head>
                <title>404 - Not Found</title>
        </head>
        <body>
                <h1>404 - Not Found</h1>
                <script type="text/javascript" src="//wpc.75674.betacdn.net/0075674/www/ec_tpm_bcon.js"></script>
        </body>
</html>

[pmon]  2020-05-05 14:07:23  error : AdvancedWebhook: Status Code: 404
```
MSTeams: not tested
Gotify: not tested

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated changelog.md with changes made

### CODE REVIEW

**Comment by DanrwAU** [comment]
Only the MSTeams plugin was mine, That one looks all good to me, i'll test it for you shortly

File: server/plugins/Gotify.json [hunk 1]
```diff
@@ -8,19 +8,9 @@
     {
       "name": "URL",
       "label": "Gotify URL (Without http/s)",
+      "description": "Gotify domain URL eq. https://example.org (Note: without /message)",
```
Comment (davidmckenzie) [inline server/plugins/Gotify.json:11]: These two lines contradict - should it include http/s or not? Also will this impact people with things already in those settings?

**Comment by DanrwAU** [comment]
Teams plugin seems to work correctly 👍

File: server/plugins/Gotify.json (hunk 1 above)
Comment (SloCompTech) [reply server/plugins/Gotify.json:11]: Oh, right, I forgot to fix this label. I will fix it.

Comment (SloCompTech) [reply server/plugins/Gotify.json]: @davidmckenzie I fixed label and added some compatibility checks, so it will work in most cases.

**Pull request merged**